core/src/main/java/com/chopshop166/chopshoplib/motors/CSSpark.java

```diff
@@ -27,12 +26,11 @@ public class CSSpark extends SmartMotorController {
      * Create a smart motor controller from an unwrapped Spark object.
      *
      * @param spark The Spark MAX/Flex oject.
-     * @param type The motor type (brushed vs brushless).
+     * @param countsPerRev The number of counts per revolution of the encoder.
      */
-    public CSSpark(final CANSparkBase spark, final MotorType type) {
-        super(new MockMotorController(),
-                new SparkEncoder(type == MotorType.kBrushless ? spark.getEncoder()
-                        : spark.getEncoder(SparkRelativeEncoder.Type.kQuadrature, 4096)));
+    protected CSSpark(final CANSparkBase spark, final int countsPerRev) {
+        super(spark, new SparkEncoder(
+                spark.getEncoder(SparkRelativeEncoder.Type.kQuadrature, countsPerRev)));
```

Comment:
CSSparkMax needs to override the encoder type to `Type.kHallSensor` with `countsPerRev` of 42 for brushless mode.